### PR TITLE
Add FAQ for artifact store remove API

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

As there is no description regarding how we can remove artifacts, I added the description to FAQ. 

## Description of the changes

I added another section for the remove methodology.

### CODE REVIEW

**Comment by eukaryo** [comment]
@not522 @c-bata Could you review this PR?

File: docs/source/reference/artifacts.rst [hunk 1]
```diff
@@ -4,7 +4,12 @@ optuna.artifacts
 ================
 
 The :mod:`~optuna.artifacts` module provides the way to manage artifacts (output files) in Optuna.
-Please note that methods defined in each ArtifactStore are not intended to be directly accessed by library users.
+
+.. note::
+   The methods defined in each ``ArtifactStore`` are not intended to be directly accessed by library users.
+
+.. note::
+   As ``ArtifactStore`` does not officially provide user API for artifact removal, please refer to `FAQ <how-can-i-delete-all-the-artifacts-uploaded-to-a-study>`_ for the hack.
```
Comment (c-bata) [inline docs/source/reference/artifacts.rst:12]: [nit] How about adding a link to the artifact tutorial as well?

File: docs/source/faq.rst [hunk 2]
```diff
@@ -714,3 +714,37 @@ Optuna may sometimes suggest parameters evaluated in the past and if you would l
...
+    Failing to do so may lead to unexpected behaviors as Optuna does not guarantee expected behaviors when users call :meth:`remove` externally.
+    Due to the Optuna software design, it is hard to officially support the delete feature and we are not planning to support this feature in the future either. 
+
+.. code-block:: python
+
+    from optuna.artifacts import get_all_artifact_meta
+
+
+    def remove_artifacts(study, artifact_store):
+        # NOTE: `artifact_store.remove` is discourged to use.
```
Comment (c-bata) [inline docs/source/faq.rst:741]:
```suggestion
        # NOTE: `artifact_store.remove` is discouraged to use.
```

File: docs/source/reference/artifacts.rst (hunk 1 above)
Comment (c-bata) [inline docs/source/reference/artifacts.rst:12]: The link to the FAQ is broken. You need to insert the custom anchor to the FAQ page. Could you fix it?

* https://optuna--5501.org.readthedocs.build/en/5501/reference/artifacts.html
* https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#use-a-custom-anchor

**Review by c-bata — CHANGES_REQUESTED** [review]
Thank you for your pull request. I left some comments.

File: docs/source/reference/artifacts.rst (hunk 1 above)
Comment (nabenabe0928) [reply docs/source/reference/artifacts.rst]: Thank you for the suggestion:) 
i applied your suggestion!

Comment (nabenabe0928) [reply docs/source/reference/artifacts.rst]: I addressed your suggestion!
Let me see how the test goes!

File: docs/source/faq.rst [hunk 3]
```diff
@@ -714,3 +714,39 @@ Optuna may sometimes suggest parameters evaluated in the past and if you would l
 
     study = optuna.create_study()
     study.optimize(objective, n_trials=100)
+
+.. _remove_for_artifact_store:
+
+How can I delete all the artifacts uploaded to a study?
+-------------------------------------------------------
+
+Optuna supports `artifacts <https://optuna.readthedocs.io/en/stable/reference/artifacts.html>`_ for large data storage during an optimization.
```
Comment (not522) [inline docs/source/faq.rst:723]:
```suggestion
Optuna supports `artifacts <https://optuna.readthedocs.io/en/stable/reference/artifacts.html>`__ for large data storage during an optimization.
```

**Review by c-bata — APPROVED** [review]
LGTM.

File: docs/source/faq.rst [hunk 4]
```diff
@@ -714,3 +714,39 @@ Optuna may sometimes suggest parameters evaluated in the past and if you would l
...
+    Failing to do so may lead to unexpected behaviors as Optuna does not guarantee expected behaviors when users call :meth:`remove` externally.
+    Due to the Optuna software design, it is hard to officially support the delete feature and we are not planning to support this feature in the future either. 
+
+.. code-block:: python
+
+    from optuna.artifacts import get_all_artifact_meta
+
+
+    def remove_artifacts(study, artifact_store):
+        # NOTE: `artifact_store.remove` is discouraged to use.
```
Comment (not522) [inline docs/source/faq.rst:743]:
```suggestion
        # NOTE: `artifact_store.remove` is discouraged to use because it is an internal feature.
```

**Review by not522 — COMMENTED** [review]
This PR is almost LGTM. Could you check my comments?

File: docs/source/faq.rst (hunk 3 above)
Comment (c-bata) [reply docs/source/faq.rst:723]: @nabenabe0928 Similar to the suggestion in  https://github.com/optuna/optuna/pull/5501#discussion_r1651961644, I think we should add a custom anchor to detect broken links. What do you think?

File: docs/source/reference/artifacts.rst [hunk 5]
```diff
@@ -4,7 +4,13 @@ optuna.artifacts
 ================
 
 The :mod:`~optuna.artifacts` module provides the way to manage artifacts (output files) in Optuna.
-Please note that methods defined in each ArtifactStore are not intended to be directly accessed by library users.
+Please check :ref:`artifact_tutorial` for the instruction.
```
Comment (nabenabe0928) [inline docs/source/reference/artifacts.rst:7]: I do not know why this line does not appear in the doc published by CI:
- https://optuna--5501.org.readthedocs.build/en/5501/reference/artifacts.html

image

Plus, the link for `remove_for_artifact_store` was not updated as well.
It appears as `FAQ` in the doc.

**Review by not522 — APPROVED** [review]
LGTM!

File: docs/source/reference/artifacts.rst (hunk 5 above)
Comment (nabenabe0928) [reply docs/source/reference/artifacts.rst:7]: Was a cache issue
It was resolved once I forced to reload it!

**Comment by nabenabe0928** [comment]
@c-bata I applied your suggestion, could you take another look?

**Review by c-bata — APPROVED** [review]
LGTM!

**Pull request merged**